src/Compilers/CSharp/Test/Emit2/Semantics/CollectionExpressionTests.cs

```diff
@@ -3340,9 +3348,17 @@ static void Main()
                     }
                 }
                 """;
-            // https://github.com/dotnet/roslyn/issues/68786: Infer nullability from collection expressions in type inference.
             var comp = CreateCompilation(source);
-            comp.VerifyEmitDiagnostics();
+            comp.VerifyEmitDiagnostics(
+                // (10,44): warning CS8619: Nullability of reference types in value of type 'IEnumerable<string?>' doesn't match target type 'IEnumerable<object>'.
```

Comment:
What is being analyzed to yield this warning? Are we actually converting the `b` to `IEnumerable<object>` before enumerating in the bound tree and warning on that?

Reply:
It looks like the comment refers to `ab = [..a, ..b]` in `TypeInference_Spread_08`.

In that test, `b` has type `IEnumerable<string?>[]` and the target collection has type `IEnumerable<object!>[]`.

`NullableWalker.VisitCollectionExpression` is reporting the warning from the implicit conversion of the *element* of the spread element expression, with type `IEnumerable<string?>`, to the *element type* `IEnumerable<object!>` of the target collection.